### PR TITLE
mold breaker refactor

### PR DESCRIPTION
**Pull request by pacmanboss256**
Mold breaker now ignores specific abilities rather than by using event checks
No longer specifies abilities that were not supposed to be ignored (isUnbreakable)

### CODE REVIEW

**Comment by pacmanboss256** [comment]
rough idea of making mold breaker only ignore certain abilities rather than not ignore certain abilities and ignore everything else

**Comment by pacmanboss256** [comment]
will fix sunsteel strike once this is merged

File: sim/battle.ts [hunk 1]
```diff
@@ -710,37 +710,10 @@ export class Battle {
 				// it's changed; call it off
 				continue;
 			}
-			if (effect.effectType === 'Ability' && !effect.isUnbreakable &&
+			if (effect.effectType === 'Ability' && effect.ignorable &&
```
Comment (urkerab) [inline sim/battle.ts:713]: Rather than ignoring `isUnbreakable` you should just remove it completely.

File: server/chat-commands/info.ts [hunk 2]
```diff
@@ -803,7 +803,7 @@ export const commands: Chat.ChatCommands = {
 						Gen: String(ability.gen) || 'CAP',
 					};
 					if (ability.isPermanent) details["&#10003; Not affected by Gastro Acid"] = "";
-					if (ability.isUnbreakable) details["&#10003; Not affected by Mold Breaker"] = "";
+					if (ability.ignorable) details["&#10003; Ignored by Mold Breaker"] = "";
```
Comment (Zarel) [inline server/chat-commands/info.ts:806]: What's with the flip? Are there more abilities affected by Mold Breaker or not affected?

Comment (Zarel) [reply server/chat-commands/info.ts:806]: (Out of the abilities that could potentially be affected by Mold Breaker.)

Comment (pacmanboss256) [reply server/chat-commands/info.ts:806]: oh its flipped because isUnbreakable will be deleted

File: data/mods/gen4/abilities.ts [hunk 3]
```diff
@@ -376,7 +377,7 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 	stickyhold: {
 		inherit: true,
 		onTakeItem(item, pokemon, source) {
-			if (this.suppressingAttackEvents(pokemon)) return;
+			if (this.suppressingAbility(pokemon)) return;
```
Comment (Marty-D) [inline data/mods/gen4/abilities.ts:380]: Here, the whole line is actually useless.

**Review by Marty-D — CHANGES_REQUESTED** [review]
- Rebound needs to be considered
- `isBreakable: false` is unnecessary on real Abilities
- Please check past gens for cases like Gen 4 Simple which override the whole effect

**Review by pacmanboss256 — COMMENTED** [review]
all set

**Comment by Marty-D** [comment]
> Please check past gens for cases like Gen 4 Simple which override the whole effect

Gen 4 Thick Fat
Gen 3 Lightning Rod (realistically this one doesn't matter, but no reason to take functionality away from Custom Games)

**Comment by Marty-D** [comment]
While reviewing I wondered if Aura Break, Dark Aura, and Fairy Aura would actually have their `Start` events suppressed in-game, so I checked and they do! But PS runs `Start` separately in `runSwitch`, so you'll need to add `suppressingAbility` to those three Abilities (kind of like how Sticky Hold did it) for completeness.

**Review by Marty-D — APPROVED** [review]
Looks good, thanks! 👍

File: server/chat-commands/info.ts (hunk 2 above)
Comment (Zarel) [reply server/chat-commands/info.ts]: I'm saying why should it be deleted?

Comment (Zarel) [reply server/chat-commands/info.ts]: I think it'd be better if it was flipped back.

**Pull request merged**